### PR TITLE
History class: create test cases ...

### PR DESCRIPTION
**Pull request by aidoxe-123**
... to improve coverage

### CODE REVIEW

**Review by Jh123x — COMMENTED** [review]
LGTM.

Just wondering why was the exception changed to an Assertion.
When we release the Jar file, assertions will not run and if there is an error there might be issues there.

**Comment by JoeyChenSmart** [comment]
Actually I think if you are using assertions, you should not re-throw them in the method

i.e.
```java
void pop() throws AssertionError {
        assert !stack.empty() : "History is empty";
        stack.pop();
    }
```
should be just
```java
void pop() {
        assert !stack.empty() : "History is empty";
        stack.pop();
    }
```
instead.

Assertions are there to allow us as developers to locate potential bugs/violations of pre conditions, so we want them to fail as quickly as possible in the code. Re throwing them like the above example will cause the location of the error to be confused (the caller of `pop` will throw the error, but we want `pop` to throw the error).


Also I think we should keep the previous (non assertion) error, for defensive programming

File: src/main/java/jimmy/mcgymmy/model/History.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ boolean empty() {
 
     /**
      * Get the previous state from history
-     * @throws EmptyStackException
+     * @throws AssertionError
```
Comment (Jh123x) [inline src/main/java/jimmy/mcgymmy/model/History.java]: I think we need to change this to EmptyStackException too.

**Review by Jh123x — CHANGES_REQUESTED** [review]
Almost LGTM ><

File: src/main/java/jimmy/mcgymmy/model/History.java (hunk 1 above)
Comment (aidoxe-123) [reply src/main/java/jimmy/mcgymmy/model/History.java]: done!

**Review by Jh123x — APPROVED** [review]
LGTM

**Pull request merged**